Setup SLES 16 Tests for HA

lilyeyes opened the pull request:
Setup SLES 16 Tests for HA: create_hdd_ha_textmode

TEAM-9960 - Setup SLES 16 Tests for HA

- Related ticket: [TEAM-9960](https://jira.suse.com/browse/TEAM-9960)
Setup SLES 16 Tests for HA
- Related job group: https://openqa.suse.de/admin/job_templates/579?
- Verification run:  https://openqa.suse.de/tests/16542765 (passed)

New VR: after revised code according comments: https://openqa.suse.de/tests/16552154#step/system_prepare/32 (passed)

mpagot commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+    zypper_call('in suseconnect-ng');
+    # De-register SLES
+    assert_script_run('suseconnect -d');
+    # Regster SLES
```
```suggestion
    # Register SLES
```

mpagot commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+    assert_script_run('SUSEConnect' . ' -p ' . 'sle-ha/' . get_required_var('VERSION') . '/' . get_required_var('ARCH') . ' -r ' . get_required_var('SCC_REGCODE_HA'));
+    # Check HA extension
+    assert_script_run('suseconnect --status-text');
+    assert_script_run('suseconnect --list-extensions');
```
Do we like to search something in the output?

lilyeyes replied:
The original purpose is just for `debug info` as SLE16 is still not good. Deleted.

Dawei-Pang commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+    select_console 'root-console';
+
+    # Just for debug purpose
+    script_run('zypper se ha | grep -i high');
```
How about checking installed pattern by "zypper patterns -i"?

lilyeyes replied:
Added `zypper patterns` not `zypper patterns -i`.

jlausuch commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+    assert_script_run('suseconnect --status-text');
+    assert_script_run('suseconnect --list-extensions');
+    # Update system
+    zypper_call('dup', timeout => 300);
```
why `dup`? I would simply do `zypper up`

jlausuch commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+
+    # Workaround for current agama installation: registration, extensions and so on
+    # Install suseconnect-ng
+    zypper_call('in suseconnect-ng');
```
It should be installed by default

lilyeyes replied:
For sles4sap it is installed by default but for sles command `suseconnect` is not found.

Amrysliu commented on tests/ha/ha_install.pm:
```diff
@@ -0,0 +1,46 @@
+    # De-register SLES
+    assert_script_run('suseconnect -d');
+    # Regster SLES
+    assert_script_run('SUSEConnect' . ' -r ' . get_required_var('SCC_REGCODE') . ' --url ' . get_required_var('SCC_URL'));
```
This part is done in system_prepare which is implemented in #21018

alvarocarvajald replied:
Agreed. Seems changes in `ha_install` are not required now that #21018 has been merged. Can you rebase and retry the VRs dropping this module?

lilyeyes replied:
Rebased, dropped ha_install.pm, added workaround for HA extension as https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21018 did not register HA extension.

lilyeyes commented:
Thanks for the comments, updated the PR according to the comments. 
  - dropped `ha_install.pm`
  - rebased with https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/21018 and added workaround for `HA extension`
  - added simple debug info after HA registration as SLE16 is till not stable 
New VR: https://openqa.suse.de/tests/16552154#step/system_prepare/32 (passed)

lilyeyes commented on tests/console/system_prepare.pm:
```diff
@@ -61,7 +61,8 @@ sub run {
             add_suseconnect_product(get_addon_fullname('serverapp'), undef, undef, undef, 300, 1);
-            add_suseconnect_product(get_addon_fullname('desktop'), undef, undef, undef, 300, 1) if is_sle('=12-sp5', get_var('ORIGIN_SYSTEM_VERSION')) && is_ppc64le;
+            add_suseconnect_product(get_addon_fullname('desktop'), undef, undef, undef, 300, 1)
+              if is_sle('=12-sp5', get_var('ORIGIN_SYSTEM_VERSION')) && is_ppc64le;
```
Just for fixing tidy issue.

jlausuch commented on tests/console/system_prepare.pm:
```diff
@@ -131,6 +132,21 @@ sub run {
+        # Regster HA extension and output some debug info for reference
+        if (get_var('SCC_ADDONS', '') eq 'ha') {
+            record_info('Regster HA extension');
+            assert_script_run('SUSEConnect' . ' -p '
```
Why not using existing util method to register addons?
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/suseconnect_register.pm#L62C5-L62C21

lilyeyes replied:
Sure, let me try `command_register()`.

lilyeyes replied:
Updated code to use command_register: https://openqa.suse.de/tests/16552983#step/system_prepare/41 (passed)

jlausuch commented on tests/console/system_prepare.pm:
```diff
@@ -131,6 +132,21 @@ sub run {
+                  . get_required_var('SCC_REGCODE_HA'));
+            # Just for debug purpose
+            zypper_call('up', timeout => 600);
+            script_run('zypper lr');
```
Just a small suggestion:
`record_info('REPOS', script_output('zypper lr -u')`
`record_info('PKGs', script_output('zypper se ha | grep -i high')`
`record_info('PATTERNS', script_output('zypper patterns')`

alvarocarvajald commented on tests/console/system_prepare.pm:
```diff
@@ -131,6 +133,17 @@ sub run {
         assert_script_run("suseconnect -r $regcode --url $regurl");
+
+        # Register HA extension and output some debug info for reference
+        if (get_var('SCC_ADDONS', '') eq 'ha') {
```
I believe this is OK for the time being, but usually `SCC_ADDONS` can contain a comma-separated list of addons, so this will probably need to be changed if other addons are available.

lilyeyes replied:
Yes. 
I added `HA workaround` between bellow lines. So the whole temporary workaround will be deleted as soon as agama support HA installation. So I did not handle `comma-separated list of addons`.

```
    ########
    # Workaround to install sle16 in beta1
...
if (get_var('SCC_ADDONS', '') eq 'ha') {
}
...
    ########
```

alvarocarvajald reviewed (approved):
LGTM

rfan1 commented:
is https://openqa.suse.de/tests/16542765 the right VR link?
